hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSinkManager.java

```diff
@@ -150,9 +151,14 @@ public synchronized void reportSinkSuccess(SinkPeer sinkPeer) {
    */
   public synchronized void chooseSinks() {
     List<ServerName> slaveAddresses = endpoint.getRegionServers();
-    Collections.shuffle(slaveAddresses, ThreadLocalRandom.current());
-    int numSinks = (int) Math.ceil(slaveAddresses.size() * ratio);
-    sinks = slaveAddresses.subList(0, numSinks);
+    if(slaveAddresses==null){
```

Comment:
Doesn't look like HBaseReplicationEndpoint ever returns null. Guarding against custom endpoint implementations?

We should expose `getRegionServers` on a base-class or interface and explicitly say that we expect a non-null answer. Follow-on..

If easy, this would be good to write a quick unit test to cover this method.

Reply:
> Doesn't look like HBaseReplicationEndpoint ever returns null. Guarding against custom endpoint implementations?

Yeah, that's indeed the case. 

>We should expose getRegionServers on a base-class or interface and explicitly say that we expect a non-null answer. Follow-on..

If easy, this would be good to write a quick unit test to cover this method.

Sounds good, actually. So we can leave `getRegionServers` at HBaseReplicationEndpoint, emphasize it should never return null and add UT.